pages/voting-guides/index.tsx

```diff
@@ -0,0 +1,149 @@
+          date: value.election.electionDate,
+          guides: [value]
+        } as VotingGuideElection;
+        result.push(electionInfo);
+      } else {
+        result[i]?.guides.push(value);
+      }
+      return result;
+    }, [] as VotingGuideElection[]);
+  } else return [] as VotingGuideElection[];
+};
+
+const VotingGuides: NextPage<{ mobileNavTitle?: string }> = ({mobileNavTitle}) => {
+
+  const user = useAuth({redirectTo:"/login"}).user; //user will not be null under this line, due to the conditional redirect
+  const votingGuidesByUserId = useVotingGuidesByUserIdQuery({userId:user?.id || ""}); 
+  const [elections, setElections] = useState<VotingGuideElection[]>([]);
+
+  useEffect(()=>{
+    const {data, isError, error} = votingGuidesByUserId;
+  if (isError){
+    alert("Error getting Voting Guides");
+    console.error("Error getting Voting Guides", error);
+  }
+  else{
+    if (data && data.votingGuidesByUserId.length) {
+      const newVotingGuides = data.votingGuidesByUserId as VotingGuideResult[];
+      const elections = getElectionsFromGuides(newVotingGuides);
+      console.log('yes data', elections);
+      setElections(elections);
+    } else {
+      votingGuidesByUserId.refetch().then(res=>{
+        const elections = getElectionsFromGuides(res.data?.votingGuidesByUserId as VotingGuideResult[]);
+        console.log('no data', elections);
+        setElections(elections);
+      });
+    }
+  };
+  },[]);
+
+  const upsertVotingGuide = useUpsertVotingGuideMutation();
+
+  const [title, setTitle] = useState("");
+  const [description, setDescription] = useState("");
+
+  return (
+    <>
+      <Head>
+        <title>Populist - Voting Guides</title>
+        <meta name="description" content="View Voting Guides." />
+      </Head>
+
+      <Layout mobileNavTitle={`${mobileNavTitle || 'Voting Guides'}`} showNavLogoOnMobile={false}>
+        <div className={styles.votingContainer}> 
+          <FlagSection title="Voting Guides">
+            {
+              elections.map(election=>(
+                <div key={election.id}>
+                  <h1 className={styles.votingHeader}>{election.date}</h1>
+                  <h2>{election.title}</h2>
+                  <p className={styles.description}>{election.description}</p>
+                    {
+                      election.guides?.map(guide=><VotingGuideCard guide={guide} key={guide.id} />)
+                    }
+                </div>
+              ))
+            }
+          </FlagSection>
+        </div>
+
+        <div className={styles.roundedCard}>
+          <h2>Add a test guide </h2>
+          <h3>(for CO Primary Election)</h3>
+          <input className={styles.guideInput} type="text" placeholder="Title" onChange={e=>setTitle(e.target.value)} value={title} />
+          <input className={styles.guideInput} type="text" placeholder="Description" onChange={e=>setDescription(e.target.value)} value={description} />
```

Comment:
We need to make an input component. 

Also I prefer named functions for event handlers - although there is no performance benefit with functional components I'd say it helps better with debugging if there's a named fn in the trace.

Reply:
for sure, extracting out a reusable input with @bastardlee style system is a must

Reply:
Agree we need an input component, but this section is purely for testing purposes and will be removed